sympy/combinatorics/free_groups.py

```diff
@@ -594,11 +594,28 @@ def commutator(self, other):
         else:
             return self.inverse()*other.inverse()*self*other
 
-    def eliminate_word(self, gen, by):
+    def eliminate_words(self, words, _all = False):
+        '''
+        Replace each subword from the dictionary `words` by words[subword].
+        If words is a list, replace the words by the identity.
+        '''
+        new = self
+        if type(words) is dict:
```

Comment:
It is usually better to write `isinstance(words, dict)` since that will work also for subclasses. (Though those may not appear in this case.)